### PR TITLE
refactor(ext/crypto): clean up rust side importKey

### PR DESCRIPTION
**Pull request by lucacasonato**
This commit cleans up the Rust side of `import_key` by using a bunch of
enums instead of structs with "type" and "data" fields.

This commit does add some duplicated code for the time being, because
a lot of the other ops still need to get the same cleanup treatment.

This is an alternative to the Rust side changes in #13027.

### CODE REVIEW

**Comment by lucacasonato** [comment]
@SeanWykes Does the pattern introduced here work for the JWK implementation?

**Comment by SeanWykes** [comment]
> @SeanWykes Does the pattern introduced here work for the JWK implementation?

Yes, should do, we'll just add two more members to KeyData enum. :)

**Comment by lucacasonato** [comment]
> Yes, should do, we'll just add two more members to KeyData enum. :)

As in two fields inside `KeyData::Jwk`, or two new enum members? If you mean the latter, what members would those be?

**Comment by SeanWykes** [comment]
> > Yes, should do, we'll just add two more members to KeyData enum. :)
> 
> As in two fields inside `KeyData::Jwk`, or two new enum members? If you mean the latter, what members would those be?

I personally find new enum members to be cleaner than having a single big `::JwK` structure.

In #12022, I use different structs and enum members for Symmetric, RSA and EC JWKs: 
`JwkSecretKey(SecretKeyComponentB64)`,  `JwkRsaKey(RSAKeyComponentsB64)` and `JwkEcKey(ECKeyComponentsB64)`.

**Comment by lucacasonato** [comment]
Makes sense. Let's cross that bridge when we get to it :-)

File: ext/crypto/import_key.rs [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+      import_key_rsassa(key_data, hash)
+    }
+    ImportKeyOptions::RsaPss { hash } => import_key_rsapss(key_data, hash),
+    ImportKeyOptions::RsaOaep { hash } => import_key_rsaoaep(key_data, hash),
+    ImportKeyOptions::Ecdsa { named_curve }
+    | ImportKeyOptions::Ecdh { named_curve } => {
+      import_key_ec(key_data, named_curve)
+    }
+  }
+}
```
Comment (bartlomieju) [inline ext/crypto/import_key.rs:69]: Why is it an async function is there are no `await` points?

Comment (lucacasonato) [reply ext/crypto/import_key.rs:69]: Because that is how it already was :shrug:. I'll go make it sync.

**Review by littledivy — APPROVED** [review]
neat

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**